packages/cli/src/cmds/agentInstaller/agentInstallerProcedure.ts

```diff
@@ -66,7 +68,36 @@ export default class AgentInstallerProcedure extends AgentProcedure {
     const appMapYml = this.configPath;
     if (!useExistingAppMapYml) {
       const initCommand = await this.installer.initCommand();
-      const { stdout } = await run(initCommand);
+      let stdout = '';
+
+      try {
+        const initCommandOutput = await run(initCommand);
+        stdout = initCommandOutput.stdout;
+      } catch (e) {
+        const error = e as Error;
+
+        if (
+          this.installer.name === 'Bundler' &&
+          error?.message.includes('but is an incompatible architecture')
+        ) {
+          await run(new CommandStruct('gem', ['uninstall', 'appmap', '-x', '--force'], this.path));
+          await run(new CommandStruct('bundle', [], this.path));
```

Comment:
If I'm understanding the problem correctly, I don't believe we can automate this for them. This error occurs because `node` is run via x64 emulation and thus the child `bundle` process is as well, causing the `appmap` dependency to be built for the wrong architecture.

The resolution `gem uninstall appmap; bundle` works for the user because there's no x64 `node` process to trigger emulation.

Reply:
Won't it depend on if the user is running VS Code natively or emulated? If they're running VS Code natively, then when we run the `bundle` command for them, it will know the architecture is `arm64`

Reply:
As @dividedmind explained it, this is due to users running an x64 version of node. The command that's failing is essentially the same as what we're running to fix it: 

https://github.com/getappmap/appmap-js/blob/main/packages/cli/src/cmds/agentInstaller/rubyAgentInstaller.ts#L68

In regard to the architecture of VS Code, I'm not sure if it's an issue. It does seem totally possible that an (old) x64 version of VS Code would spawn terminals through Rosetta, but I'm unaware of how prevalent this may be.